### PR TITLE
Re-introduce Operate OpenAPI explorer!

### LINKED ISSUE #3114: Re-introduce Operate API Explorer
#2944 caused unexpected UI issues, and was reverted almost immediately in https://github.com/camunda/camunda-docs/pull/3072

See #3072 for details on the UI issues. They were caused by [a known issue in the plugin I used for implementation](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/issues/626).

There are two possible paths forward, both of which bring uncertainty: 

1. Identify the cause of the bug in the plugin, and attempt to resolve it.
2. Re-implement the explorer with a different plugin.

I'd start with the first option, with a time-box, before moving on to the second.

### PR DESCRIPTION
**Pull request by pepopowitz**
Resolves #3114!

## Description

Re-introduces the Operate OpenAPI explorer!

We had to revert it after the initial implementation, due to unexpected UI regressions. This version does not wreck the UI 💪 

## Why won't it wreck the UI this time?

As described [in the plugin's issue](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/issues/626#issuecomment-1882349730), a seemingly unrelated dependency update (@mdx-js/react) fixes the UI regressions.

## What it looks like

image

## Preview site

https://loquacious-pika-a05ac2.netlify.app/

## How does the user get to the Operate API explorer?

Sidebar navigation options are added to the `next` and `8.4` versions, in the Operate REST API sections.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [ ] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/apis-tools/operate-api/authentication.md
Comment (pepopowitz) [inline docs/apis-tools/operate-api/authentication.md:1]: @christinaausley re: your review, I'm most interested in how you feel about my extraction of the Authentication docs, and the related adjustments I made in the API overview.

File: package.json [hunk 1]
```diff
@@ -22,15 +22,20 @@
...
-    "test:watch": "jest --watch"
+    "test:watch": "jest --watch",
+    "api:generate": "npm run docusaurus gen-api-docs",
+    "api:generate:operate": "npm run docusaurus gen-api-docs operate"
   },
   "dependencies": {
     "@auth0/auth0-react": "^2.2.1",
     "@docusaurus/core": "^2.4.1",
     "@docusaurus/preset-classic": "^2.4.1",
+    "@mdx-js/react": "^1.6.22",
```
Comment (pepopowitz) [inline package.json:33]: This was 1.6.21 on the failed attempt.

Comment (akeller) [reply package.json:33]: Thank you for this comment!

**Comment by pepopowitz** [comment]
heads up @akeller @christinaausley that I published a preview site for this: https://loquacious-pika-a05ac2.netlify.app/

**Comment by akeller** [comment]
While reviewing locally and in your preview site, I see all the languages are enabled. We've talked about priority languages (Java/Spring, Node, C#), but I would like to know if we poll folks internally about the language set what they will prefer here.

I saw no issues with images or annotations across the docs. I looked at a few random Component, Self-Managed, Reference, and Guide pages across 8.3, 8.4, and `/next/`. Prettier is obviously complaining, but I don't see anything blocking!

File: docusaurus.config.js [hunk 2]
```diff
@@ -239,6 +275,48 @@ module.exports = {
...
+        variants: ["requests", "http.client"],
+      },
+      {
+        highlight: "csharp",
+        language: "csharp",
+        logoClass: "csharp",
+        variant: "RestSharp",
+        variants: ["restsharp", "httpclient"],
+      },
+    ],
```
Comment (pepopowitz) [inline docusaurus.config.js:319]: @akeller here are the definitions of languages for sample code. I'm happy to change any of this if you have requests! I attempted to sort them based on what I understood to be popularity in the original implementation of this.

**Comment by pepopowitz** [comment]
> While reviewing locally and in your preview site, I see all the languages are enabled. We've talked about priority languages (Java/Spring, Node, C#), but I would like to know if we poll folks internally about the language set what they will prefer here.
> 
> I saw no issues with images or annotations across the docs. I looked at a few random Component, Self-Managed, Reference, and Guide pages across 8.3, 8.4, and `/next/`. Prettier is obviously complaining, but I don't see anything blocking!

(see previous comment)

File: README.md [hunk 3]
```diff
@@ -19,6 +19,7 @@ Did you know Camunda Engineers are the primary documentation authors?
 - [Features](./howtos/markdown-and-mdx-features.md)
 - [Technical writing cheatsheet](./howtos/technical-writing-cheatsheet.md)
 - [Technical writing style guide](./howtos/technical-writing-styleguide.md)
+- [Interactve API Explorers](./howtos/interactive-api-explorers.md)
```
Comment (pepopowitz) [inline README.md:22]:
```suggestion
- [Interactive API Explorers](./howtos/interactive-api-explorers.md)
```

File: README.md [hunk 4]
```diff
@@ -30,3 +31,4 @@ Documentation strategy and infrastructure is owned by the Developer Experience t
 - [Features](./howtos/markdown-and-mdx-features.md)
 - [Versioning](./howtos/versioning.md)
 - [Release Procedure](./howtos/release-procedure.md)
+- [Interactve API Explorers](./howtos/interactive-api-explorers.md)
```
Comment (pepopowitz) [inline README.md:34]:
```suggestion
- [Interactive API Explorers](./howtos/interactive-api-explorers.md)
```

File: docusaurus.config.js (hunk 2 above)
Comment (akeller) [reply docusaurus.config.js:319]: Can you swap csharp and go?

Comment (pepopowitz) [reply docusaurus.config.js]: resolved in latest commit

**Review by akeller — APPROVED** [review]
🟢  from me, but I see there is still an open Q for @christinaausley to review

File: docs/apis-tools/operate-api/authentication.md
Comment (pepopowitz) [reply docs/apis-tools/operate-api/authentication.md:1]: I'm going to merge as-is, if you have any feedback afterward I'm happy to incorporate it.

Comment (christinaausley) [reply docs/apis-tools/operate-api/authentication.md:1]: I am good with this 👍

**Pull request merged**